### PR TITLE
[expo-image] Fix blurhash not working and causing fetch to `blurhash:` uri.

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

We broke Blurhash support in expo-image on web in https://github.com/expo/expo/pull/27251.

# How

Now on web we always append the `blurhash:/` prefix and strip it in decode function to match thumbhash.

In addition, we also avoid passing a raw blurhash string to an  tag to avoid a fetch that would error out previously and render a broken image icon.

# Test Plan

Tested NCL in expo go web. This PR doesn't touch non-web files (only scoped) so it should be fairly safe to merge.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-image/CHANGELOG.md [hunk 1]
```diff
@@ -10,7 +10,8 @@
 
 ### 🐛 Bug fixes
 
-- Fixed an issue where `isBlurhashString` always returned `true` .([#27251](https://github.com/expo/expo/pull/27251) by [@alanjhughes](https://github.com/alanjhughes))
+- [Web] Fix blurhash not working and causing fetches to `blurhash:/` uri.
+- Fixed an issue where `isBlurhashString` always returned `true`. ([#27251](https://github.com/expo/expo/pull/27251) by [@alanjhughes](https://github.com/alanjhughes))
```
Comment (aleqsio) [inline packages/expo-image/CHANGELOG.md:14]: moved the dot at the end of sentence :)

File: packages/expo-image/src/web/hooks.ts [hunk 2]
```diff
@@ -13,19 +13,24 @@ export function useThumbhash(source: ImageSource | null | undefined) {
     () => (isThumbhash ? { uri: thumbHashStringToDataURL(strippedThumbhashString) } : null),
     [strippedThumbhashString, isThumbhash]
   );
-  return thumbhashSource;
+  return useMemo(() => [thumbhashSource, isThumbhash] as const, [thumbhashSource, isThumbhash]);
```
[the remark this answers is not in the document]
Comment (aleqsio) [reply packages/expo-image/src/web/hooks.ts:16]: isThumbhash can be computed synchronously, but thumbhashSource is computed asynchronously, and is null until it's computed.

[the remark this answers is not in the document]
Comment (tsapeta) [reply packages/expo-image/src/web/hooks.ts:16]: Maybe I don't get something, but `const thumbhashSource = useMemo(...)` is synchronous 😅 

If `useMemo` only returns its own dependencies, isn't it equivalent to just returning the dependencies?

File: packages/expo-image/src/utils/blurhash/useBlurhash.tsx [hunk 3]
```diff
@@ -22,14 +22,16 @@ export function useBlurhash(
   punch = punch || 1;
 
   const [uri, setUri] = useState<string | null>(null);
+  const isBlurhash = (!!blurhash && !!blurhash.uri && isBlurhashString(blurhash.uri)) ?? false;
 
   useEffect(() => {
     let isCanceled = false;
 
-    if (!blurhash || !blurhash.uri || !isBlurhashString(blurhash.uri)) return;
+    if (!blurhash || !blurhash.uri || !isBlurhash) return;
```
[the remark this answers is not in the document]
Comment (aleqsio) [reply packages/expo-image/src/utils/blurhash/useBlurhash.tsx]: this is mostly for TS to ensure that `strippedBlurhashString` is not optional  – let's keep this typeguard `if` if possible.

**Comment by Jonatthu** [comment]
@aleqsio any updates on this? How can I workaround this on the mean time?

**Review by lukmccall — APPROVED** [review]

**Pull request merged**